sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs

```diff
@@ -135,6 +135,28 @@ public void CreateSkillsetReturnsCorrectDefinitionOcrShaper()
             });
         }
 
+        [Fact]
+        public void CreateSkillsetReturnsCorrectDefinitionShaperWithNestedInputs()
+        {
+            Run(() =>
+            {
+                SearchServiceClient searchClient = Data.GetSearchServiceClient();
+                CreateAndValidateSkillset(searchClient, CreateTestSkillsetWithNestedInputs());
+            });
+        }
+
+        [Fact]
+        public void CreateSkillsetThrowsExceptionWithNonShaperSkillWithNestedInputs()
```

Comment:
Be careful not to fall into the trap of testing the service instead of testing the SDK. For SDK purposes, it's sufficient to test one `CloudException` error path per operation.

Reply:
True in general, but I think because tests serve as documentation to some extent as well, I thought it'd be useful to have this here, just to reinforce the fact that a "runtime" exception could occur if non shaper skillsets have recursive/nested inputs

Reply:
Sure, but remember that 99.9% of the time these tests run against recorded mock responses. So if the service behavior changes, you may not notice that the test still passes until the next time you run it in Record mode. That's burned us before.